desc/backend.py

```diff
@@ -555,6 +556,45 @@ def fun_vmap(fun_inputs):
 
         return fun_vmap
 
+    def scan(f, init, xs, length=None, reverse=False, unroll=1):
```

Comment:
I am assuming this is probably not vectorized.
So if we want to integrate an array of shape (L, M, N) along the last axis, will this work?

Reply:
You'd need to transpose the array. This function is basically just a non-jax version of a jax function (for legacy reasons we've been trying to keep things compatible with pure numpy), so this is just trying to match the API of the jax version. In practice this never gets used, only the jax version `jax.lax.scan` (imported above)

Reply:
Haven't we discussed not supporting regular numpy anymore? I would be in favor of that in for a new release